contracts/TokenRequest.sol

```diff
@@ -6,50 +6,70 @@ import "@aragon/os/contracts/common/SafeERC20.sol";
 import "@aragon/os/contracts/lib/token/ERC20.sol";
 import "@aragon/apps-token-manager/contracts/TokenManager.sol";
 import "./lib/UintArrayLib.sol";
+import "./lib/ArrayUtils.sol";
 
 /**
 * Expected use requires the FINALISE_TOKEN_REQUEST_ROLE permission be given exclusively to a forwarder. A user can then
 * request tokens by calling createTokenRequest() to deposit funds and then calling finaliseTokenRequest() which will be called
 * via the forwarder if forwarding is successful, minting the user tokens.
 */
-contract TokenRequest is AragonApp {
+  contract TokenRequest is EtherTokenConstant, AragonApp {
 
     using SafeERC20 for ERC20;
     using UintArrayLib for uint256[];
+    using ArrayUtils for address[];
 
     bytes32 constant public SET_TOKEN_MANAGER_ROLE = keccak256("SET_TOKEN_MANAGER_ROLE");
     bytes32 constant public SET_VAULT_ROLE = keccak256("SET_VAULT_ROLE");
     bytes32 constant public FINALISE_TOKEN_REQUEST_ROLE = keccak256("FINALISE_TOKEN_REQUEST_ROLE");
+    bytes32 constant public ADD_TOKEN_ROLE = keccak256("ADD_TOKEN_ROLE");
+    bytes32 constant public REMOVE_TOKEN_ROLE = keccak256("REMOVE_TOKEN_ROLE");
+    bytes32 constant public SUBMIT_TOKEN_REQUEST_ROLE = keccak256("SUBMIT_TOKEN_REQUEST_ROLE");
+    bytes32 constant public SET_TIME_TO_EXPIRY_ROLE = keccak256("SET_TIME_TO_EXPIRY_ROLE");
 
     string private constant ERROR_NO_AMOUNT = "TOKEN_REQUEST_NO_AMOUNT";
     string private constant ERROR_NOT_OWNER = "TOKEN_REQUEST_NOT_OWNER";
     string private constant ERROR_NO_DEPOSIT = "TOKEN_REQUEST_NO_DEPOSIT";
     string private constant ERROR_ETH_VALUE_MISMATCH = "TOKEN_REQUEST_ETH_VALUE_MISMATCH";
     string private constant ERROR_TOKEN_TRANSFER_REVERTED = "TOKEN_REQUEST_TOKEN_TRANSFER_REVERTED";
+    string private constant ERROR_CANNOT_ADD_TOKEN_MANAGER = "TOKEN_REQUEST_CANNOT_ADD_TOKEN_MANAGER";
+    string private constant ERROR_TOKEN_ALREADY_ADDED = "TOKEN_REQUEST_TOKEN_ALREADY_ADDED";
+    string private constant ERROR_TOKEN_NOT_CONTRACT = "TOKEN_REQUEST_TOKEN_NOT_CONTRACT";
+    string private constant ERROR_TOKEN_NOT_EXIST = "TOKEN_REQUEST_ERROR_TOKEN_NOT_EXIST";
 
     struct TokenRequest {
         address requesterAddress;
         address depositToken;
         uint256 depositAmount;
         uint256 requestAmount;
+        uint64 date;
     }
 
     TokenManager public tokenManager;
     address public vault;
 
+    mapping(address => bool) public tokenAdded;
+    address[] public acceptedTokenList;
+
+    uint256 public timeToExpiry;
```

Comment:
Why have you added the above 3 vars, what are they used for? Because they aren't currently used to modify the behaviour of the contract.